binheap.py

```diff
@@ -0,0 +1,79 @@
+class BinaryHeap:
+    def __init__(self, bin_type, values=None):
+        self._storage = []
+        self._size = 0
+        self.bin_type = bin_type
```

Comment:
self.bin_type --> self._bin_type, This is also an implementation detail that shouldn't be used outside the class.